erc20/vyper/vyper-erc20-spec.ini

```diff
@@ -71,56 +74,66 @@ requires:
     andBool 0 <=Int BAL_TO    andBool BAL_TO    <Int (2 ^Int 256)
 
 [transfer-success]
-k: #execute => #halt
 statusCode: _ => EVMC_SUCCESS
 output: _ => #asByteStackInWidth(1, 32)
 log: _:List ( .List => ListItem(#abiEventLog(ACCT_ID, "Transfer", #indexed(#address(CALLER_ID)), #indexed(#address(TO_ID)), #uint256(VALUE))) )
 
 [transfer-success-1]
-storage:
-    #hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID) |-> (BAL_FROM => BAL_FROM -Int VALUE)
-    #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)     |-> (BAL_TO   => BAL_TO   +Int VALUE)
-    _:Map
+storage: M1:IMap => M2:IMap
+origStorage: M1
 +requires:
+    andBool select(M1, #hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID)) ==Int BAL_FROM
+    andBool select(M1, #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)) ==Int BAL_TO
     andBool CALLER_ID =/=Int TO_ID
     andBool VALUE <=Int BAL_FROM
-    andBool BAL_TO +Int VALUE <Int (2 ^Int 256)
-
+    andBool BAL_TO +Int VALUE <Int (2 ^Int 256)  
+ensures:
+    ensures select(M2, #hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID)) ==Int BAL_FROM -Int VALUE
+    andBool select(M2, #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)) ==Int BAL_TO +Int VALUE
+    andBool M1 ==IMap M2 except
+      (SetItem(#hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID))
+       SetItem(#hashedLocation({COMPILER}, {_BALANCES}, TO_ID)) .Set)
+    
 [transfer-success-2]
-storage:
-    #hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID) |-> BAL_FROM
-    _:Map
+storage: M:IMap
+origStorage: M
 +requires:
+    andBool select(M, #hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID)) ==Int BAL_FROM
     andBool CALLER_ID ==Int TO_ID
     andBool VALUE <=Int BAL_FROM
+ensures:
 
 [transfer-failure]
-k: #execute => #halt
 statusCode: _ => EVMC_REVERT
 output: _ => _
 log: _
 
 [transfer-failure-1]
-storage:
-    #hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID) |-> (BAL_FROM => _)
-    #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)     |->  BAL_TO
-    _:Map
+storage: M1:IMap => M2:IMap
+origStorage: M1
 +requires:
+    andBool select(M1, #hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID)) ==Int BAL_FROM
+    andBool select(M1, #hashedLocation({COMPILER}, {_BALANCES}, TO_ID)) ==Int BAL_TO
     andBool CALLER_ID =/=Int TO_ID
     andBool ( VALUE >Int BAL_FROM
      orBool   BAL_TO +Int VALUE >=Int (2 ^Int 256) )
-
+ensures:
+    ensures M1 ==IMap M2 
+        except (SetItem(#hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID)) .Set)
+   
 [transfer-failure-2]
-storage:
-    #hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID) |-> BAL_FROM
-    _:Map
+storage: M1:IMap => M2:IMap
+origStorage: M1
 +requires:
+    andBool select(M1, #hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID)) ==Int BAL_FROM
     andBool CALLER_ID ==Int TO_ID
     andBool VALUE >Int BAL_FROM
+ensures:
+    ensures M1 ==IMap M2 
+        except (SetItem(#hashedLocation({COMPILER}, {_BALANCES}, CALLER_ID)) .Set)
```

Comment:
why do you need this `except` here? the original spec says that the storage does NOT update at all.